third_party/terraform/utils/config.go.erb

```diff
@@ -289,7 +290,8 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	c.clientContainerBeta.UserAgent = userAgent
 	c.clientContainerBeta.BasePath = containerBetaClientBasePath
 
-	dnsClientBasePath := c.DNSBasePath + "projects/"
+	dnsClientBasePath := removeBasePathVersion(c.DNSBasePath)
+	dnsClientBasePath = strings.ReplaceAll(dnsClientBasePath, "/dns/", "")
```

Comment:
I chose to manipulate the existing base paths here instead of changing the base path to make MM-based resources work with no modifications. Only one resource (record set) and 2 data sources use this client library